go/cmd/dolt/commands/sqlserver/sqlserver.go

```diff
@@ -245,6 +244,11 @@ func StartServer(ctx context.Context, versionStr, commandStr string, args []stri
 		return err
 	}
 
+	err = generateYamlConfigIfNone(ap, help, args, dEnv, serverConfig)
```

Comment:
This should only run in the case that an explicit value for --config wasn't provided

Reply:
The code you have now will correctly not overwrite a file named config.yaml, but it should never attempt to write a file if a config file was provided. Plz add a test for this too

Reply:
The bats test "config file isn't generated if a config is specified in args" should cover this (it gives a config file not named config.yaml and makes sure a new config.yaml isn't generated).